### DISCUSSION TITLE
Craft 4 to 5 upgrade, Symfony Mailer error on deployment

### DISCUSSION THREAD

**Discussion by emilyo-cogapp** [craftcms/cms#15342] [Q&A]
Hello, I just upgraded from Craft 4 to 5 and when deploying the site to our staging environment I got this error in `storage/logs/console.log`: `Symfony\Component\Mailer\Transport\Dsn::__construct(): Argument #2 ($host) must be of type string, null given` and the frontend was erroring. 

The site's hosted on Platform.sh so I was able to fix it by enabling their 'allow outgoing mail' setting as this sets the `$host` variable I believe, but previously we had left this blank and had no errors when deploying. 

I'm just trying to figure out why this error might have been thrown _now_ and not before, so I'm wondering whether there's something to do with a change in either how Craft 5 implements email or even an update in the Symfony mailer or something else along the line to cause this to now throw an error? Any ideas appreciated 😅

**Comment by brandonkelly** [comment]
Craft 5 upgraded [yiisoft/yii2-symfonymailer](https://github.com/yiisoft/yii2-symfonymailer) from v2 to v4 (#14239), so there could be some breaking changes as part of that, although it’s not noted in the [changelog](https://github.com/yiisoft/yii2-symfonymailer/blob/master/CHANGELOG.md).

Which specific transport adapter are you using?

**Reply by emilyo-cogapp** [reply]
So I’ve done some investigating and we’re using SMTP, which I can see does require a hostname value in the CMS. We have this set to an environment variable called `PLATFORM_SMTP_HOST` which I didn’t set up initially on this project but I assume comes from Platform.sh as that’s what the variable should be called there and which locally we have included in the ddev .env file, but with no value e.g.: `PLATFORM_SMTP_HOST=`. So we don’t have email set up and working locally, but haven’t needed it so this hasn’t affected local development.

On Platform.sh we haven’t added this environment variable manually anywhere, but when ‘Enable outgoing emails’ is set, I can grep the variable and see it has the correct value. On production this is enabled and the email settings are correct and email sending works, but on staging this was not enabled (ever) and grepping the variable returns nothing at all (hence the value returning ‘null’ in the error I guess), but when I enable it and the variable has a value, the error goes away and the build is able to finish. 

I can see that if we’ve set Craft up to use SMTP and set the hostname to an environment var, it makes sense that not having this set on a particular environment might cause issues, but the thing is that we’ve never had this set on our staging environment but it’s only errored and failed the build since the upgrade to Craft 5. The other thing is that Platform.sh defaults to having ‘Enable outgoing emails’ set to ‘on’ for production environments and ‘off’ for others: https://docs.platform.sh/development/email.html#1-turn-on-outgoing-email so it seems to me like a) in order to stop this erroring for Craft we have to go against the generally advised/default behaviour on Platform.sh, which I guess is probably fine if necessary but also feels like something we shouldn’t necessarily _have_ to do? and b) this is a newly introduced (if niche) issue that’s been introduced with Craft 5 and might trip other people up who might have similar setups?

**Reply by brandonkelly** [reply]
Thanks for the info. I believe I just got this fixed for the next release (750e53c98df09d864a4b64afd3c69d9a0e2ad195). You can test by changing your `craftcms/cms` requirement in composer.json to:

```json
"craftcms/cms": "5.x-dev as 5.2.6",
```

and running `composer update`.

**Reply by emilyo-cogapp** [reply]
🎉  That works, thank you!

**Reply by brandonkelly** [reply]
Thanks for testing!

**Reply by brandonkelly** [reply]
Craft 5.2.7 is out with that fix. Thanks again.
